### PR TITLE
Enable more Python types to be supported by the DALI python function

### PR DESCRIPTION
**Pull request by JanuszL**
- adds support to DALI to and from DLPack conversion
- extends types support for fn.ones, fn.full to include
  int8, uin32, uint64
- adds tests

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- adds support to DALI to and from DLPack conversion
- extends types support for fn.ones, fn.full to include
  int8, uin32, uint64
- adds tests

### Affected modules and functionalities:
- DALI_TYPE_SWITCH_WITH_FP16
- to and from DLPack conversion

### Key points relevant for the review:
- NA

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
    - test_python_function.test_different_types
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: dali/pipeline/data/types.h [hunk 1]
```diff
@@ -636,6 +636,12 @@ inline std::ostream &operator<<(std::ostream &os, DALIDataType dtype) {
         {__VA_ARGS__}                                \
       }                                              \
       break;                                         \
+    case DALI_INT8:                                  \
```
Comment (mzient) [inline dali/pipeline/data/types.h:639]: Food for thought:

This macro is used only once - I'd remove it, add some definition of tensor-compatible types and use `TYPE_SWITCH` instead.
```C++
#define DALI_INTEGRAL_TYPES uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t
#define DALI_NUMERIC_TYPES DALI_INTEGRAL_TYPES, float, double
#define DALI_NUMERIC_TYPES_FP16 DALI_NUMERIC_TYPES, float16
```

File: include/dali/core/common.h [hunk 2]
```diff
@@ -49,11 +49,12 @@ using std::vector;
...
 using uint16 = uint16_t;
-using int16 = int16_t;
-using int64 = int64_t;
+using uint32 = uint32_t;
 using uint64 = uint64_t;
+using int8 = int8_t;
+using int16 = int16_t;
 using int32 = int32_t;
-using uint32 = uint32_t;
+using int64 = int64_t;
```
Comment (mzient) [inline include/dali/core/common.h:57]: Can we remove those instead? They're not used that much and many places in the code use the standard names anyway.

**Review by mzient — APPROVED** [review]
Approved, but I'm not happy about extending proliferating vestigial features.

File: include/dali/core/common.h (hunk 2 above)
Comment (JanuszL) [reply include/dali/core/common.h]: Removed all the types and converted all the occurrence to `*_t`.

File: dali/pipeline/data/types.h (hunk 1 above)
Comment (JanuszL) [reply dali/pipeline/data/types.h]: Done

**Comment by JanuszL** [comment]
!build

**Review by mzient — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Pull request merged**